projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.secrets.controller;
+
+import com.vmware.taurus.controlplane.model.api.DataJobsSecretsApi;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Map;
+
+@RestController
+@ComponentScan(basePackages = "com.vmware.taurus.secrets")
+@Tag(name = "Data Jobs Secrets")
+public class DataJobsSecretsController implements DataJobsSecretsApi {
+  static Logger log = LoggerFactory.getLogger(DataJobsSecretsController.class);
+
+  @Override
+  public ResponseEntity<Void> dataJobSecretsUpdate(
+      String teamName, String jobName, String deploymentId, Map<String, Object> requestBody) {
+    log.debug("Updating secrets for job: {}", jobName);
+    throw new ResponseStatusException(
+        HttpStatus.NOT_IMPLEMENTED, "Secrets service is not implemented");
+  }
+
+  @Override
+  public ResponseEntity<Map<String, Object>> dataJobSecretsRead(
+      String teamName, String jobName, String deploymentId) {
+    log.debug("Reading secrets for job: {}", jobName);
+    throw new ResponseStatusException(
+        HttpStatus.NOT_IMPLEMENTED, "Secrets service is not implemented");
+  }
+}
```

Comment:
In addition to the questions raised, may elaborate on the design decision to use an "empty" controller? What an empty controller is, and what is its purpose? Is that an incremental development step rather than a design decision on extensibility?

Reply:
Check the first sentence of the PR ;)

If everything goes according to plan I'll add the actual implementation next week, but in the meantime we need the updated client libraries so that we can start working on the python sdk changes in parallel.

Reply:
It's an incremental change.
Properties API provides strong extensibility leveraging `IPropertiesServiceClient`. Is the secrets API designed to be extensible?

Reply:
It's going to have the same capabilities as the properties api/service.